src/constants.js

```diff
@@ -141,7 +141,18 @@ export const getSearchParams = () => {
     .getAll('tags')?.[0]
     ?.split?.(',')
     .reduce?.(tagsMapper, []);
-  const operatingSystem = searchParams.getAll('operating_system');
+  const operatingSystem = searchParams
+    .getAll('operating_system')
+    .map((osFilter) => {
+      if (typeof osFilter !== 'object') {
+        const index = osFilter.search(/\d/);
+        const osName = osFilter.substring(0, index);
+        const osVersion = osFilter.substring(index);
+        const [major] = osVersion.split('.');
+        const groupName = `${osName} ${major}`;
```

Comment:
Could this maybe be accomplished with a RegEx and match the parts needed instead of splicing and dicing the the strings?

Reply:
hope it helps
```suggestion
  const operatingSystem = searchParams
    .getAll('operating_system')
    .map((osFilter) => {
      if (typeof osFilter !== 'object') {
        const match = osFilter.match(/([a-zA-Z\s]+) (\d+\.\d+)/);
        if (match) {
          const [osName, osVersion] = match;
          const [major] = osVersion.split('.');
          const groupName = `${osName} ${major}`;
          return { osName, value: osVersion, groupName };
        }
      }
    });

Reply:
@vcwild Oh man!!! I never saw this suggestion. I ended up solving it another way, which also worked. :)